vendor/bat-native-ads/src/bat/ads/internal/ads_impl.cc

```diff
@@ -347,6 +367,82 @@ void AdsImpl::SetConfirmationsIsReady(const bool is_ready) {
   is_confirmations_ready_ = is_ready;
 }
 
+std::map<std::string, std::vector<AdsHistory>> AdsImpl::GetAdsHistory() {
+  std::map<std::string, std::vector<AdsHistory>> ads_history;
+  base::Time now = base::Time::Now().LocalMidnight();
+
+  auto ad_history_details = client_->GetAdsShownHistory();
+  for (auto& detail_item : ad_history_details) {
+    // FIXME(erogul): This should probably be done in the JavaScript
```

Comment:
@jasonrsadler Thinking it would make more sense to perform this truncation on the JavaScript side, but let me know.

Reply:
Agreed as this can then be platform specific